converse.js

```diff
@@ -634,11 +635,35 @@
             },this), 200));
         };
 
+        this.ping = function (jid, timeout, success, error){
+            if (typeof jid === 'undefined'     || jid == null)     { jid = this.bare_jid; }
+                if (typeof timeout === 'undefined' || timeout == null) { timeout = 45; }
+                if (typeof success === 'undefined' ) { success = null; }
+                if (typeof error === 'undefined' ) { error = null; }
+                if (this.connection) this.connection.ping.ping( jid, success, error, timeout );
+        };
+
+        this.pong = function (ping){
+		    converse.connection.ping.pong(ping);
+	        return true;
+    	};
+
+	    this.registerPongHandler = function (){
+			converse.connection.ping.addPingHandler( this.pong );
+    	};
+
+	    this.registerPingHandler = function (){
+            if (this.ping_interval>0){
+		        window.setInterval( function() { converse.ping(); }, this.ping_interval*1000);
+                }
+    	};
```

Comment:
With this code the server will be pinged every `ping_interval` amount of seconds. Even when there is already lots of other stanzas being sent out between converse.js and the server.

An enhancement might be to rather wait `ping_interval` seconds from the last stanza sent to the server (no matter what stanza that was). This avoids pinging the server when it's not really necessary (i.e. when there's already lots of traffic between client and server).

I'm however not yet sure how we could go about doing that.